runtime/compiler/runtime/MetaDataDebug.cpp

```diff
@@ -558,14 +558,19 @@ TR_Debug::printByteCodeStack(int32_t parentStackIndex, uint16_t byteCodeIndex, c
       void *bcPrintFunc = (void *)jitBytecodePrintFunction;
       if (parentStackIndex == -1)
          {
-         sprintf(indent, " \\\\");
+         int len = snprintf(indent, bufLen, " \\\\");
+         if ((0 < len) && (len < bufLen)) {
+            bufLen -= len;
+         } else {
+            bufLen = 0;
+         }
```

Comment:
This should be simplified to use `strcpy()` when `bufLen > 3`
```c
         if (bufLen > 3)
            {
            strcpy(indent, " \\\\");
            bufLen -= 3;
            }
         else
            {
            bufLen = 0;
            }
```

Reply:
We shouldn't be using strcpy but I've updated to use strncpy. Doesn't hard coding numbers violate some coding standards? I can update it again to use sizeof if you agree.

Reply:
Perhaps it would be better if `" \\\\"` and its length were captured in locals:
```c
         static const char indentString[] = " \\\\";
         size_t indentSize = sizeof(indentString) - 1;
         if (bufLen > indentSize)
            {
            strncpy(indent, indentString, indentSize + 1);
            bufLen -= indentSize;
            }
         else
            {
            bufLen = 0;
            }
```
The code that replaces line 587 can be similar.

Reply:
I don't see any reason to set `bufLen = 0` so I didn't include that part (even though it was in my original change).

Reply:
It occurs to me that rather than allocating space for the indent string, it could be expressed as just its width. In `bcdump.c` where the string would be printed, that width could be paired with the `"%*s" format and an empty string. For clarity:
```c
    printf("'%*d'\n", 6, "");
```
prints  six spaces between a pair of single-quotes.

It's unclear to me whether the special case `" \\\\"` is at the beginning or the end, but either way is easily handled at the one place `indent` is actually used (line 169 of `bcdump.c`).
```c
		outputFunction(userData, "%s%5i %s ", indent, pc, JavaBCNames[bc]);
```
would become (assuming `\\ ` should appear at the beginning):
```c
		outputFunction(userData, "\\\\ %*s%5i %s ", width, "", pc, JavaBCNames[bc]);
```

Reply:
I don't know it's that simple for the `\\`. It's recursive and passing in `site._byteCodeInfo.getCallerIndex()`. It's beyond what I'm trying to do, which is to remove sprintf. It's debug code, so it's not normally active and doesn't need to be optimal.

Reply:
I guess I have nothing better to do. I found a jitdump file and it contains sections like the following. I've made the changes.
```
 \\ java/lang/Access.getCarrierThreadLocal(Ljdk/internal/misc/CarrierThreadLocal;)Ljava/lang/Object;
 \\    4 JBinvokevirtual 139 java/lang/ThreadLocal.getCarrierThreadLocal()Ljava/lang/Object;
 \\      25 JBinvokespecial 16 java/lang/ThreadLocal.get(Ljava/lang/Thread;)Ljava/lang/Object;
 \\          2 JBinvokevirtual 22 java/lang/ThreadLocal.getMap(Ljava/lang/Thread;)Ljava/lang/ThreadLocal$ThreadLocalMap;
```

Reply:
I can confirm that JIT logs look the same before and after this change.